### ISSUE TITLE
RFC: Design and UX ideas

### ISSUE THREAD

**Issue by zadam** [zadam/trilium#1614]
Hi all,

First beta of 0.46 is nearing, and I'm starting to think about 0.47. One of the focus areas will be the look and feel of the app which is currently ... let's say less than ideal.

What I'd like to do:

*   make the UI more visually attractive - nicer, sprinkle in some colors. This would mean a new default theme, probably with light and dark variants. Current theme could be preserved as "classic theme".
*   make the app easier to use with the leverage of visual cues. Current theme is visually quite minimalistic, e.g. many widgets are not visually separated which might make the app more difficult to understand.
*   create sort of visual language for common elements like primary buttons, help buttons etc.
*   add new theme extension points (CSS variables) leveraging the more consistent visual language mentioned above. Currently theming is little more than a different color scheme and one must (ab)use unstable CSS classes to change the look more significantly. Of course the theming "API" will never be complete but we can do better than now.
*   change app icon - current icon is too complex. Simpler design with some colors would look better. Decent inspiration can be found on [google](https://www.google.com/search?q=trillium+ontario&client=ubuntu&hs=Ie3&channel=fs&sxsrf=ALeKk01vQ2QpqyYqSHUyXFcVc0OIjK9NCA:1612606965344&source=lnms&tbm=isch&sa=X&ved=2ahUKEwjAm9OKhdXuAhWJHOwKHYNkBe8Q_AUoAXoECBUQAw&biw=1698&bih=871). Flower/leaf/plant based motif should remain ...

As part of this there might be some smaller functional/layout UI changes, but larger UI reorganization should remain out of scope.

Now all this is pretty difficult task for me - I'm not a designer and my abilities in this area are very limited so I thought I might ask here for input. Things I'm interested in:

*   specific ideas - e.g. what you think might look better or might be easier to visually identify
*   inspiration links and screenshots - do you particularly like some app and its design? Post here a screenshot and ideally comment on why you think it looks/works well. A cool example is e.g. [Allure](https://github.com/JadeVane/Allure):

![](https://user-images.githubusercontent.com/617641/107115606-1b618800-686e-11eb-941a-75cf1410eca1.png)

*   are there perhaps designers working with open source projects. Some (limited) monetary compensation might be possible. (applies also for icon design)
*   to people developing their own themes - what extension points would you like to be added into CSS variables?

**Comment by nil0x42** [comment]
Hi ! Trilium is an awesome tool but people decide to start testing a note taking app based on what they see.

I like the idea of removing the horizontal pane below tabs like in the screenshot :)

Being able to hide the left pane's actions menu is also something i would add as an option (keep it for newcommers, but after 1 week using trilium shortcuts are easily memorized, so this pane becomes distracting:
![Screenshot_2021-02-08_10-35-38](https://user-images.githubusercontent.com/3504393/107208585-b9ce2480-69f9-11eb-8f46-51d5c26eed88.png)


Another suggestion i have, is to have a clear distinction of current note in `link map` section. Current styling difference is not obvious enough.

**Comment by zadam** [comment]
> Being able to hide the left pane's actions menu is also something i would add as an option (keep it for newcommers, but after 1 week using trilium shortcuts are easily memorized, so this pane becomes distracting:

This is already removed in master (0.46). Thanks for the other points!

**Comment by thefirebird00** [comment]
I would love to see the Allure theme as the new default :smiley: 
I use it myself and - all the small bugs aside - it super awesome.

Keep up the good work! :partying_face:

**Comment by davialexandre** [comment]
Hello! First of all, I wanted to say that I've been using Trilium daily for a few weeks and I love it! Thank you for this awesome application!

In terms of UX and Design, I'm a big fan of minimalist interfaces, so many of my suggestions are about hiding stuff. I'm also a big fan of the Notion interface, so some of the suggestions are inspired by it. Here they are:

- Make it possible to disable or hide the tabs. I use the web version and never used tabs, as I prefer to simply open a new tab/window on my browser
- Move the note paths widget from the sidebar to the top of the center pane (where applications usually place their breadcrumb component). This would make it possible so see the full to notes inside deep branches
- Remove the "buttons" bar (not sure what is the correct name for this). Maybe move the most useful ones (like search and new note) to the top right site, like the Allure theme does):
![2021-02-11-194337_938x51_scrot](https://user-images.githubusercontent.com/388373/107708934-1738ce80-6ca3-11eb-9abe-6b239156f57e.png)
- Hide root from the tree view (I don't think it's possible to have more than one root, is it?)
- Add more white space between elements, to improve readability. For example, I have a custom css that:
  - Adds some left/right padding to the sidebar
  - Adds a lot of left and right padding to the note details container
  - Increases the default line-height of paragraphs and list items
- Differentiate the sidebar from the content:
  - Use a different background color for each (usually sidebars are darker than the content)
  - Use a smaller font-size for the elements in tree (right now the size is the same as the note content)
- Increase the font-size and font-weight of the title of the currently open note to make it really stand-out
- Make attributes first-class citizens. Basically, make something like this:
![2021-02-11-193504_1403x188_scrot](https://user-images.githubusercontent.com/388373/107709108-6c74e000-6ca3-11eb-8b7d-2558e7c980de.png)

To look more like this (screenshot taken from Notion):
![2021-02-11-193559_564x435_scrot](https://user-images.githubusercontent.com/388373/107709138-7e568300-6ca3-11eb-91d2-9b93c55fa34e.png)
- Display the note icon next to the title of the currently open note (like Notion does):

![2021-02-11-193921_838x250_scrot](https://user-images.githubusercontent.com/388373/107709197-9a5a2480-6ca3-11eb-9e99-6e06587d318c.png)

- Make it easier to change the note icon. Perhaps by making the displaying a list of all the available icons when clicking on the icon (also like Notion does):

![notion-icons](https://user-images.githubusercontent.com/388373/107709324-c70e3c00-6ca3-11eb-905b-aa1f281b5685.gif)

**Comment by egalanos** [comment]
This is quite a big topic! Some basic suggestions based on using the 0.45.10 desktop release below.

Are you going to experiment on master or plan on a new branch? It can be a bit difficult to know without being able to easily see and experiment as changes are made.

### Top level

1. Menu cleanup
    - [ ] Remove word 'Menu' and downwards triangle
    - [ ] Move to the top right
 2. Lines separating sidebars and navigation section
    - [ ] Remove fading effect (It just confuses things)

### Navigation section

1. 'Jump to note' removal
   - [ ] Add functionality as part of interactive search results
 2. 'Recent changes' removal
    - [ ] Add as a special dynamic note. This would be more usable than a pop-up.
 3. 'Enter protected session' cleanup
    - [ ] Change Protect icon from shield to Padlock icon as users already understand it. 
    - [ ] Protected session status icon moved to top right just left of main menu icon.
    - [ ] Leave protected session icon could be then a combination of a padlock and hourglass perhaps?

### Tree pane

#### Top action icons

1. Collapse note tree button removal
   - It doesn't seem important enough to need to be there?
 2. Scroll to active note removal?
    - It doesn't seem important enough to need to be there as you would only not see the active note if you manually collapsed a parent?
  3. Search move
     - [ ] Move to navigation section of UI. Perhaps above where note actions live

#### Active note branch path

1. Dropdown for note locations removal
   - [ ] This doesn't seem important enough to always be there. Information could be moved into note info section.

#### Tree pane

1. Root note icon cleanup
   - [ ] Remove >> as it does nothing and is confusing as it suggests show/hide like the sidebar show/hide icons
 2. Tree settings icon location
    - [ ] The options could probably be moved under the main menu options as the two things listed don't need to be frequently changed
  3. Text notes icon simplification
     - [ ] Perhaps a plain square with lines similar to the Google docs icon within Google drive would be an improvement?
  4. Tree expansion arrow improvement
     - [ ] A smaller version would look better as the current size is too prominent

### Note view

1. Note name font sizing
    - [ ] Increase size to be a title size as it is currently smaller than Heading 2 size
 2. Attributes line cleanup
    - [ ] Should be made thicker for a more prominent separation between note title and rest of note
    - [ ] Hide attribute summary when there are 0 attributes
  3. Type dropdown auto hide
     - [ ] Should this be part of the info or action part of note once a note has content?
   4. Note actions dropdown
      - [ ] Rename to 'Actions'
      - [ ] Remove the word note from the actions (eg 'Note source' becomes 'Source')

### Note info sidebar

1. Make it a per note button in top right to the right of actions (the Google drive information icon is a good example)

**Comment by Grymas** [comment]
All great suggestions, Personally I'd love for attributes to be a bit more controllable in their display :

As @davialexandre shows, attributes could be displayed in a more readable way, possibly to the right sidebar (in Note Info or maybe a separate category such as Note Attributes).

Note hover display could use some options, such a optionally showing only the note title (without the path), and optionally not display the note attributes.

The left sidebar might benefit from a tighter look (tons of notes in a lot of branches with long titles make it hard to navigate and require it to be wide) and being more readable, but I'm not sure what the best approach is here without potentially making it too cluttered or inversely too spread-out.

**Comment by oxdc** [comment]
![allurefocus](https://user-images.githubusercontent.com/29519076/108205924-a8ef8280-7160-11eb-8cc3-274ddc87ab39.png)

Inspired by [Allure](https://github.com/JadeVane/Allure) and [mochi](https://app.mochi.cards/):

![mochi](https://user-images.githubusercontent.com/29519076/108206101-e3f1b600-7160-11eb-944f-8d8dd92018f6.png)

**Comment by zadam** [comment]
Hi, thanks for all the input! Although I can't react to all the comments I do read them and they provide a lot of food for thought. I've already incorporated some of that into 0.46 (currently in beta).

@oxdc 

> It would be nice to export notes as a `.zip` file preserving all attributes for distributing and installing extensions or themes.

Export to ZIP does that. But when importing you must uncheck "safe import", otherwise some attributes are mangled for security reasons.

**Comment by thfrei** [comment]
Hi zadam. Thank you for your effort!

**General Design:**

As a former long time OneNote user, I try to simulate its design. 
I used the melon theme and adapted it, so that the content is more prominent and all non content elements go in the background, by using darker colors. Also, the tree is more dense.

![grafik](https://user-images.githubusercontent.com/7283497/108872526-8b0b9d00-75fa-11eb-94b1-c4b0e8a50dfc.png)

Usually I disable the right sidebar, since I do not need the details. In the theme it looks like this:

![grafik](https://user-images.githubusercontent.com/7283497/108872634-a5457b00-75fa-11eb-9b9d-0eac26890bf2.png)

**Workspaces and Outlook**

I find the concept of workspaces great. For me the concept of having some "folders" and notes inside is simple. And the workspaces could "simulate" this.

Personally, I am most used to OneNote and still think, that it makes navigating really clear. The colors also help to distinguish. They are auto-generated, but changeable.

![grafik](https://user-images.githubusercontent.com/7283497/108873639-9e6b3800-75fb-11eb-987e-8b2e79efd992.png)

**Comment by thfrei** [comment]
One addition, what I also use a lot are labels for my journal:
![grafik](https://user-images.githubusercontent.com/7283497/108874799-cc9d4780-75fc-11eb-9641-8fc26570317c.png)

I think it would be more tidy, if the labels were in a more grid-like view.

These labels I use for example in a sleep-diary:
![grafik](https://user-images.githubusercontent.com/7283497/108874621-9cee3f80-75fc-11eb-8d8f-ce773104dd1b.png)

**Comment by zhaiyujin** [comment]
I like trilium very much. I often use it to write text notes. Unfortunately, code notes can be highlighted, but text notes can't. If text notes can have a navigation directory, that would be perfect for me

**Comment by sbhmdmshah** [comment]
Theme Idea
The theme titled "Gruvbox Material Dark" is very beautiful. Would look better with a black background. If we could do something similar 😁 
![Screenshot from 2021-03-01 23-30-58](https://user-images.githubusercontent.com/65122647/109538765-7b062880-7ae6-11eb-88cf-70ae573dcc00.png)

**Comment by derbirch** [comment]
I made a moderated version of icon face to fit in bigsur according to Apple's guidline
original 👇
<img width="150" alt="截屏2021-03-11 14 50 15" src="https://user-images.githubusercontent.com/33411000/110747478-690c4e80-8279-11eb-97ab-5812e1064ad3.png">
moderated 👇
<img width="148" alt="截屏2021-03-11 14 50 54" src="https://user-images.githubusercontent.com/33411000/110747485-6c073f00-8279-11eb-977d-f8518b93d018.png">

And I upload the file if anyone wants to try.
[Icon.icns.zip](https://github.com/zadam/trilium/files/6120940/Icon.icns.zip)

If you still need someone to help with the new icon, I am glad to make some sketches.😊

**Comment by zadam** [comment]
@derbirch 

> If you still need someone to help with the new icon, I am glad to make some sketches.:blush:

Hi, that would be appreciated.

One thing which I don't like about the current icon is that it's too "complicated", too detailed. Something simpler would be preferred. It does not have to represent Trillium flower, but I'd like to keep nature motif, e.g. just a leave. This time I'm not against color, but can be B&W too. So any sketches are welcome :)

**Comment by derbirch** [comment]
> @derbirch 
> 
> > If you still need someone to help with the new icon, I am glad to make some sketches.:blush:
> 
> Hi, that would be appreciated.
> 
> One thing which I don't like about the current icon is that it's too "complicated", too detailed. Something simpler would be preferred. It does not have to represent Trillium flower, but I'd like to keep nature motif, e.g. just a leaf. This time I'm not against color, but can be B&W too. So any sketches are welcome :)

Cool, I would post it here once I have any sketches 🙃

**Comment by JadeVane** [comment]
Hi, there

I'm the author of the Allure theme, and let me tell something about that.

About several months ago, I have been using tirlium for a month and always felt that the default interface of trilium is not good and clear enough. After searching for a whole, I found that there are only a few themes that can be used at the moment, and most of the themes have a great degree of improvement, so I decidde to spend some time making a theme by myself for a better experience.

When the Allure theme is about to be completed, I have asked some questions [here](https://github.com/zadam/trilium/issues/1576) to get help to finalize the theme. I think somebody might like the Allure theme because it has received praise from @zadam , so I decided to publish this topic to the github project.

But in fact, I did not receive any feedback, except for a few stars, after the release of the Allure theme. (maybe some colors are not used well, or there is a part that can be improved, but there is no way to get that.) So this makes me think that maybe nobody care much about its appearance like me, so there is no need to maintain the allure project.

I still spent some time improving the allure theme according to my own ideas, but it has not been released for the time being because I have no motivation to write related instructions, maybe you can take a look at some details of the latest development of the Allure theme

![image](https://user-images.githubusercontent.com/34258753/113254732-89b84900-92f9-11eb-9a16-ccffa5cc65c2.png)

But now the reply below this discussion makes me feel that it is necessary to continue to maintain this theme, and it make me feel happy that the Allure theme has generated such a big profit.

However, as I have mentioned, if we want to design a better theme for trilium, there are too many things that need to be improved. This involves a huge amount of work and time, but anyway, I will try to make some suggestions to make Trilium better:

- **[fix the position of the edit button](https://github.com/zadam/trilium/issues/1788)**
- **put the quick search box at the top of the note list**
so that it can fit the width of the note list, and hide and display with the note list at the same time
- **add a fixed class to the `widget-title` of the right pane**
- **add a fixed class for each displayed div**
for example, the child element div of #root-widget, instead of just using the `.component` class that most divs have

Maybe there are some problems that I encountered in the process of making themes, but I haven't remembered them for the time being. I will continue to add here when I remember them again.

**Comment by davialexandre** [comment]
@JadeVane Just wanted to say I love the "more radical" version of the theme! It's exactly I have in mind and I think of something more minimalist

**Comment by blumpcus** [comment]
@JadeVane 
I just wanted to say that I also really like your theme and would be very happy to see it continue being developed.
In fact, since @zadam mentioned that he liked it, it would be a beautiful default theme, or at least an option as one of the default themes. I'm not sure if you have any interest at all but I would love to see a night/dark version of the same theme.
Thanks again for spending your time working on this theme, it's one of the most polished and mature themes that I've seen for Trilium.

**Comment by JadeVane** [comment]
> @davialexandre : It's exactly I have in mind and I think of something more minimalist

Thanks for your appreciation. In fact, I removed the note tabs to make the interface more concise. I don’t know if this will cause performance problems if you open too many tabs because Allure-clear does not provide a button to close the tabs. So I cannot guarantee that this will not get error before Trilium provides a switch to disable the tabs.

<br>

> @blumpcus : I would love to see a night/dark version of the same theme

Yeah, night mode may be a good suggestion, so what about this

![image](https://user-images.githubusercontent.com/34258753/113407327-a0d16680-93df-11eb-9066-8002095db19e.png)

Or maybe need to make the color of the entire interface darker? feel free to tell your feeling.

**Comment by oxdc** [comment]
> Or maybe need to make the color of the entire interface darker?

It would be nice with higher contrast.

> Or take a look at this more radical version, named Allure-clear.

I really hope `Allure-clear` could be the default theme. But the tabs are somewhat important !! The best option is to hide the tab bar by default, and show it when multiple tabs are opened.

**Comment by Mangiola** [comment]
> So I cannot guarantee that this will not get error before Trilium provides a switch to disable the tabs.

Last time I tried making a theme with no tabs, there ended up being 10+ background notes open because I kept accidentally control-clicking notes in the tree without noticing. Definitely think there should be a way to disable tabs, or at least the creation of new ones.

I too really like the Allure-clear option. The complete lack of a horizontal menu at the top makes it so much easier to look at.

**Comment by blumpcus** [comment]
@JadeVane 
This looks stunning! I love it. I would say that it's definitely my favorite look in Trilium thus far.
Thank you for such a quick reply, I'm very excited about this.

I hope that both your themes are developed (Allure & Allure-clear) as I see them both being very useful.
Personally, I work a lot with tabs so I couldn't use Allure-clear without some sort of tab integration, but the main Allure theme (especially the dark one you just displayed) would be a delight.

Thanks again.

**Comment by zadam** [comment]
Hi, I'll mention again that I'm very impressed by your work @JadeVane, your designs are visually very attractive and already provide a lot of food for thought. I can see one of your designs become the default theme of Trilium, if license permits.

I have some concerns regarding usability - the designs look very minimalistic and clean, but OTOH I think there might be some usability problems regarding e.g. not easily detectable buttons (they often look as normal text) and missing borders marking different parts of UI. That's not necessarily a problem as long as we provide some alternative, more "usable" but perhaps less beautiful, theme(s).

Another thing I would mention is that I consider tab support in the built-in themes as must have since there isn't a way to disable tabs.

So far you provided several designs which have big differences in how elements are laid out, I assume you used some CSS positioning wizardry etc. to achieve such radical changes to the app layout. I believe that we should pick one such design and adjust the actual HTML layout to it so that such "hacks" are not necessary. In theory Trilium could provide several different looking layouts, but especially for the beginning let's pick one and focus on that one first.

I'm thinking also about backwards compatibility. If we redesign the layout then it will affect also other existing themes which might cause some issues for some users. This needs some thought.

In terms of work and planning - I did not actually start on this yet. I'm working currently on 0.47 which contains mostly smaller features and once that is done I'd like to start actively working on the redesign.

**Comment by JadeVane** [comment]
> but OTOH I think there might be some usability problems regarding e.g. not easily detectable buttons (they often look as normal text) and missing borders marking different parts of UI.

In fact, I think the visual priority of the buttons in the current interface is enough. This is what a button should be that you can easily find it when you need and won’t be attracted by the style of the button when you need to focus on the content.

<br>

> such "hacks" are not necessary. In theory Trilium could provide several different looking layouts, but especially for the beginning let's pick one and focus on that one first.

It really took me too much time to solve some style problems by "hacks", maybe even more than half of the development time, It is indeed necessary to make some changes for this.

<br>

> let's pick one and focus on that one first.

`Allure-clear` (`Allure-night` of course) is the focus of my current work, I also prefer this more "clear" version.

<br>

> the tabs are somewhat important

> Definitely think there should be a way to disable tabs, or at least the creation of new ones.

> I work a lot with tabs so I couldn't use Allure-clear without some sort of tab integration

> Another thing I would mention is that I consider tab support in the built-in themes as must have since there isn't a way to disable tabs.

I think you all misunderstood me on this point that tabs do not conflict with the "clear" of `Allure-clear`, as I said before, "according to my own ideas", I remove the tabs because it is not necessary for me, I streamlined everything that can be streamlined according to my own standards, so it is feasible to use tabs on `Allure-clear`, just learn more below

![image](https://user-images.githubusercontent.com/34258753/113470946-b308de80-948b-11eb-92fa-fb625091aeee.png)

> It would be nice with higher contrast.

yes, higher contrast make it better.

![image](https://user-images.githubusercontent.com/34258753/113470932-8fde2f00-948b-11eb-93b1-dcccd3124782.png)

<br>

> I really hope Allure-clear could be the default theme. 

>  I can see one of your designs become the default theme of Trilium, if license permits.

My pleasure. :smile:

**Comment by chee** [comment]
How about something like this for the icon: 
![two three pointed stars laid on top of eachother](https://user-images.githubusercontent.com/178266/115127605-46332f80-9fc7-11eb-8f87-70311768e26f.png)
except, you know, better

but, like, two 3-pointed stars on top of each other is a lot less complex while still evoking a trillium

**Comment by chee** [comment]
@JadeVane `allure-clear` is looking so lovely

**Comment by zadam** [comment]
@JadeVane do you have the CSS for allure-clear and would you share it (ideally both dark and light variants)? Or are these mockups only?

**Comment by JadeVane** [comment]
@zadam Yes, css are already in use, not just mockups. Recently I was thinking about pushing the update to github, but I was so busy that I forgot to do this, so sorry about that.

But in fact, I didn’t push the update immediately because I encountered some problems. I hope to solve these problems when the new version of trilium is updated, instead of letting everyone use a bug version. I originally planned to have more discussions with you to solve these problems, but I forgot to do that.

I made a GIF to let you know more about the bug I said:

![Allure-night-demo](https://user-images.githubusercontent.com/34258753/115146778-96bf9100-a08a-11eb-8738-0a51093d78da.gif)

These bugs include:
- The search bar cannot be collapsed
- The position of the tab will block other content
- Some of the content is difficult to capture to adapt to these styles (I am not very familiar with programming css)

In addition, there are some problems that are not bugs but look bad:
- Edit button position is too obtrusive
- The attr information can actually be placed in the information bar on the right, so that you can read the content more attentively

Will there be any changes in the latest version of trilium that can help us solve these problems?

@chee thanks a lot for your donation, and this really motivates me!

**Comment by chee** [comment]
> @chee thanks a lot for your donation, and this really motivates me!

no problem! i'm really looking forward to trying out allure-clear !!  :purple_heart:

**Comment by zadam** [comment]
> Will there be any changes in the latest version of trilium that can help us solve these problems?

My plan is to essentially rebuild the structure and potentially also some features around the allure-clear in Trilium 0.48. This is not simple though and will take time (probably a few months until stable release). As a result the theme CSS will be probably radically simplified.

**Comment by JadeVane** [comment]
Anyway, the latest version of the `Allure-clear` theme (and `Allure-night` ) has been updated, there may still be some minor issues, but `Allure-clear` is worth a try. but there is one thing you need to pay attention to when using the allure-clear theme, 

**Do NOT keep more than one version of css style to trilium, or the styles of several css style will affect each other and cause unexpected errors.**

Hope it can be solved in the next version of trilium.

**Comment by chee** [comment]
> **Do NOT keep more than one version of css style to trilium, or the styles of several css theme will affect each other and cause unexpected errors.**
> 
> Hope it can be solved in the next version of trilium.

yes, trilium should probably only load in the stylesheet of the _active_ `#appCss` note

**Comment by oxdc** [comment]
Another suggestion: CSS styles of addons, such as similar notes panel, are hard coded in JS or DOM. It would be more flexiable if they were separated as editable files.

**Comment by chee** [comment]
maybe a nice book icon would work:

![tril-color](https://user-images.githubusercontent.com/178266/115600435-3a4bb400-a2cc-11eb-8911-4b630aec8f94.png)


![image](https://user-images.githubusercontent.com/178266/115595289-45034a80-a2c6-11eb-897c-c47ee4df0eaf.png)

**Comment by zadam** [comment]
> Do NOT keep more than one version of css style to trilium, or the styles of several css theme will affect each other and cause unexpected errors.

This is not really technically solvable. CSS stylesheets are global and mix to each other.

The way this should be approached is that every theme should use a prefix selector `body.theme-MY_THEME` - when this pattern is used then the different themes don't interact with each other.

**Comment by chee** [comment]
> This is not really technically solvable. CSS stylesheets are global and mix to each other.

Sure it is.
Only inject a `<link>` into the dom for the user's current theme. When the theme changes, remove that theme's `<link>` from the dom and inject the new one.

Alternately: set a `media` attribute on the `<link>` tags for inactive themes to something that will never match, such as `disabled`. When the theme changes, remove the `media` attribute from the selected theme and set it on the others.

**Comment by chee** [comment]
I have some thoughts here: 
![image](https://user-images.githubusercontent.com/178266/115798353-96deca00-a3c5-11eb-95a5-8b63603f97fa.png)

- I think it's good if the tabs are only in the center pane (where the viewed note is), rather than being over the whole app. That's the place the tabs have an effect.
- I like moving the toolbar over to the left hand rail, before #left-pane, and using the text of the buttons as `title` instead.
- The tabs showing note icons is also good.

**Comment by zadam** [comment]
> I think it's good if the tabs are only in the center pane (where the viewed note is), rather than being over the whole app. That's the place the tabs have an effect.

That's actually incorrect, tab switch (can) influence the whole screen. It's just that the left pane typically does not change much (but can e.g. with per-tab hoisting). Right pane changes per-tab too.

But I think this is rather a design question.

> I like moving the toolbar over to the left hand rail, before #left-pane, and using the text of the buttons as title instead.

Yes, I agree. Same as in Allure or e.g. in VS Code.

> The tabs showing note icons is also good.

Possibly. Currently tabs show icon when hoisted into workspaces, but not otherwise (to make workspaced tabs more prominent)

**Comment by chee** [comment]
> That's actually incorrect, tab switch (can) influence the whole screen. It's just that the left pane typically does not change much (but can e.g. with per-tab hoisting). Right pane changes per-tab too.

oh yeah! that's true. in that case: I don't know why it feels right this way, but it sure does (to me)

**Comment by chee** [comment]
> > That's actually incorrect, tab switch (can) influence the whole screen. It's just that the left pane typically does not change much (but can e.g. with per-tab hoisting). Right pane changes per-tab too.
> 
> oh yeah! that's true. in that case: I don't know why it feels right this way, but it sure does (to me)

i think it's because: the tabs, and both sidepanes. they feel like things that annotate and operate on the note view. they are each independent views and controls on the note pane. you know what i mean?
like, conceptually, the tabs aren't changing the side bars. the tabs are changing the note, and the sidebars change because they are views on the active note.

**Comment by derbirch** [comment]
Human, here I am again, sorry for my late update, kind of busy recently. And I made a version below.
![trilium_Ico_Design_ICON](https://user-images.githubusercontent.com/33411000/117207107-2785c300-ae26-11eb-83e8-471e4eb24a30.png)
![trilium_Ico_Design_ICON_TEXT_VERTICAL](https://user-images.githubusercontent.com/33411000/117207122-2d7ba400-ae26-11eb-8d09-cd2a94741428.png)
![trilium_Ico_Design_ICON_TEXT_HORIZON](https://user-images.githubusercontent.com/33411000/117207130-2eacd100-ae26-11eb-8165-b6212ed18256.png)

**Comment by zadam** [comment]
@derbirch not bad! Thanks!

**Comment by derbirch** [comment]
> @derbirch not bad! Thanks!

Cool, glad to help

**Comment by JadeVane** [comment]
I’m trying something new recently, and I think most of them worth considering although some of the changes may be radical.

![image](https://user-images.githubusercontent.com/34258753/117262590-0489fc00-ae84-11eb-8c56-14b2a9ea6611.png)

This does not mean that tabs should be removed, but it is necessary for considering moving more buttons and functions to the left or right pane so that we can focus more on reading and editing.

**Comment by zadam** [comment]
I can report first progress with the re-design:

![image](https://user-images.githubusercontent.com/617641/119241276-8309bd80-bb55-11eb-8367-78886a003524.png)

So far I've been working mostly on the layout changes:

* the top button row was moved to the left button column, saving vertical space
* the whole right pane with widgets was removed. I was never able to make it fully useful and in most cases it just took place. Custom widgets should still be able to position themselves on the right.
* instead of the right pane it's now possible to open 2 (or more) notes side by side
* I'd like to make the functionality from right side widgets available in the form of buttons under/next to the note title

So far the changes were mostly layout related and I did not spend much time on the looks department.

**Comment by blumpcus** [comment]
@zadam 
I love the direction this is going in.
Thanks for the thoughtful choices thus far.

**Comment by raphwriter** [comment]
Having notes side by side is a great feature, awesome! :) Maybe there could even be a shortcut analog to "open in new tab" such as "open left/right"?

Also, I have a suggestion/idea for a new logo: The idea is that the leaves resemble the letter "T" (for trilium)
![grafik](https://user-images.githubusercontent.com/28687075/120345949-395e6700-c2fb-11eb-87ac-8564edebdee7.png)

**Comment by zadam** [comment]
@raphwriter that's actually pretty awesome! Is that your work?

**Comment by raphwriter** [comment]
Thanks I am glad you like it! Yes it's my work! I haven't checked yet if
there are similar logos out there (definitely possible since it is a
relatively common shape, I guess).
I'd be happy to keep working on it some more. For instance I realised that
there maybe should be a variant including the Trilium lettering ;)

zadam ***@***.***> schrieb am Di., 1. Juni 2021, 20:44:

**Comment by zadam** [comment]
Great, I like even this original variant, but seeing more would be of curse great :-)

Perhaps one small suggestion would be to try out also some different shades of green... For example, I like the coloring in these icons https://github.com/zadam/trilium/issues/1614#issuecomment-832996869

**Comment by zadam** [comment]
Some progress on the dark theme:

![image](https://user-images.githubusercontent.com/617641/120541456-31312500-c3ea-11eb-8ecc-82facab58ca9.png)

Most of the widgets from the right sidebar are now consolidated into the button bar under the title.

**Comment by raphwriter** [comment]
Some progress on the logos as well!

Here are some refined designs. I felt that the previous version was not 'compact' enough so I, well, compacted them.  😄 
Also I applied a greener color scheme, I think it works quite well.

![grafik](https://user-images.githubusercontent.com/28687075/120543709-e9f86380-c3ec-11eb-99f5-03f5f0daa2a1.png)

**Comment by omaidb** [comment]
> 我又来了，抱歉更新晚了，最近有点忙。我在下面做了一个版本。
> ![trilium_Ico_Design_ICON](https://user-images.githubusercontent.com/33411000/117207107-2785c300-ae26-11eb-83e8-471e4eb24a30.png)
> ![trilium_Ico_Design_ICON_TEXT_VERTICAL](https://user-images.githubusercontent.com/33411000/117207122-2d7ba400-ae26-11eb-8d09-cd2a94741428.png)
> ![trilium_Ico_Design_ICON_TEXT_HORIZON](https://user-images.githubusercontent.com/33411000/117207130-2eacd100-ae26-11eb-8165-b6212ed18256.png)

真漂亮

**Comment by meichthys** [comment]
> 
> 
> Some progress on the logos as well!
> 
> Here are some refined designs. I felt that the previous version was not 'compact' enough so I, well, compacted them. 😄
> Also I applied a greener color scheme, I think it works quite well.
> 
> ![grafik](https://user-images.githubusercontent.com/28687075/120543709-e9f86380-c3ec-11eb-99f5-03f5f0daa2a1.png)

Just my opinion, but i reallly like the one in the bottom right!

**Comment by omega3** [comment]
Instead of similar note area I would prefer to have area that shows all labels that are in a note. And in my opinion Trilium lacks label pane where all labels are listed in alphabetic order. 

Also I'd like to be able to disable in options attributes line.

**Comment by meichthys** [comment]
@Y-con I believe syntax highlighting is already supported. You just need to be using a code note and then select the type of code you are using from the top right menu.￼

**Comment by Y-con** [comment]
@meichthys Yep,thanks for the remind,it do work.But there are cases that we need the abilities of the Text Note like "CheckBox" ,"ToDoList" etc as well as the code block with the ability of Code Note's syntax highlighting.
So the limitation is
1.code block in Text Note can't support syntax highlighting
2.or maybe the Code Note can't support Text Note abilities(weird)

**Comment by zadam** [comment]
It's not that weird actually. There are richt text editors and code editors (with highlighting). Is there a rich text editor which can also do code highlighting? I'm not aware of any ...

**Comment by zhaiyujin** [comment]
![1628959043(1)](https://user-images.githubusercontent.com/22185888/129453232-dedfe602-482a-4579-899f-a7e5638fb6c8.png)

**Comment by meichthys** [comment]
@zhaiyujin I could be misunderstanding your question, but I think you are looking for the back-tick code formatting:

Like this: `code here`
typed like: \`code here\`

**Comment by paul1149** [comment]
I'm on .47,7. The main thing keeping me from using Trillium is the available themes. I often use Dark themes, but the contrast with the #fff font is brutal on the eyes. Similar is true of the White theme, though a bit less because the background has a tinge of shade to it. But to my eyes it's not enough softening.

Also, different colors for the tree pane would soften the impact of the whole window.

Those two principles - lessening the contrast and providing different colors for major interface elements, soften the impact on the eyes. 

Adding a choice of font face would also be a huge advancement. This is especially true when you're cross-platform, because fonts render differently on different OSes. 

The SteelBlue theme is interesting, but I actually find its contrast too subtle. And the monotone background across all elements is still a problem. And for me, there is the added problem of the serif font, which I find very hard to read.

I'm attaching a shot of my CherryTree install. I didn't spend a lot of time on it, but it does illustrate something I find pleasant to look at.

I'm going to check out the Allure theme you've cited.

![cherrytree shot](https://user-images.githubusercontent.com/13952246/133657241-e5368b9d-eb16-4367-bce3-62df5d07e2cb.png)

**Comment by raphwriter** [comment]
@paul1149 not to toot my own horn but you could also check out https://github.com/raphwriter/trilium-theme-melon 
maybe you like it

**Comment by paul1149** [comment]
Thank you, @raphwriter . I did notice that, and I think it's tastefully done. And it's certainly fine to mention it here.  But I'm looking to break up the large white background across the whole UI. I have a neurological condition which makes doing so very advantageous.

**Comment by JadeVane** [comment]
@paul1149 

> But I'm looking to break up the large white background across the whole UI. I have a neurological condition which makes doing so very advantageous.

I'm not sure of the true meaning of "break up the large white background across the whole UI", but I tried to make some changes, hoping to come in handy.

Any feedback is welcome.

![2021-09-18_11-55](https://user-images.githubusercontent.com/34258753/133871937-99502d8f-262b-4b14-ae2a-b3f2320f67ae.png)

**Comment by omega3** [comment]
I added idea about being able to open cloze databases. 
https://github.com/zadam/trilium/issues/2172

This would greatly enhance possibilities of information management in Trillium.

**Comment by paul1149** [comment]
I think that shaded tree on the left helps give the interface some pleasing subtlety.

**Comment by yllekz** [comment]
I would love if there could be some kind of theme that replicates the OneNote UI and fonts. I generally really like the OneNote look and it would be cool if there could be a mockup of it done in Trilium to ease the transition.

**Comment by zadam** [comment]
This is an old issue, please open a new one if you have some proposal.

**Comment by romaric-siri** [comment]
> @paul1149
> 
> > But I'm looking to break up the large white background across the whole UI. I have a neurological condition which makes doing so very advantageous.
> 
> I'm not sure of the true meaning of "break up the large white background across the whole UI", but I tried to make some changes, hoping to come in handy.
> 
> Any feedback is welcome.
> 
> ![2021-09-18_11-55](https://user-images.githubusercontent.com/34258753/133871937-99502d8f-262b-4b14-ae2a-b3f2320f67ae.png)

Have you released this one yet ?

**Comment by leo-google-custom-search** [comment]
> One addition, what I also use a lot are labels for my journal: ![grafik](https://user-images.githubusercontent.com/7283497/108874799-cc9d4780-75fc-11eb-9641-8fc26570317c.png)
> 
> I think it would be more tidy, if the labels were in a more grid-like view.
> 
> These labels I use for example in a sleep-diary: ![grafik](https://user-images.githubusercontent.com/7283497/108874621-9cee3f80-75fc-11eb-8d8f-ce773104dd1b.png)





https://lnkr.fun/8b51567
